sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -1583,3 +1586,104 @@ def get_trial_metrics(
             )
 
         return observation_logs.observation_log.metric_logs
+
+    def get_job_logs(
+        self,
+        name: str,  # Experiment Name
+        trial_name: str,  # Trial Name
+        follow: Optional[bool] = False,
+        namespace: Optional[str] = None,
+    ) -> Iterator[str]:
+        """
+        Get logs from a Trial.
+
+        Args:
+            name: Name of the Katib experiment
+            trial_name: Name of the trial
+            follow: Whether to follow the log stream (similar to kubectl logs -f)
+            namespace: Kubernetes namespace. If None, uses the client's namespace
+            container: Container name. If None, uses the first container
+
+        Returns:
+            Iterator of log lines as strings
+
+        Raises:
+            RuntimeError: If the trial or job is not found, or if there's an API error
+            ValueError: If required parameters are missing or invalid
+        """
+        if not name:
+            raise ValueError("Experiment name is required")
+
+        if not trial_name:
+            raise ValueError("Trial name is required")
+
+        if namespace is None:
+            namespace = self.namespace
+
+        try:
+            # Get the trial to find the associated job
+            trial = self.custom_api.get_namespaced_custom_object(
+                group=constants.KUBEFLOW_GROUP,
+                version=constants.KATIB_VERSION,
+                namespace=namespace,
+                plural=constants.TRIAL_PLURAL,
+                name=trial_name,
+            )
+
+            # Extract job name from trial status
+            if "status" not in trial or "conditions" not in trial["status"]:
+                raise RuntimeError(f"Trial {trial_name} not found or has no status")
+
+            # The job name is typically the same as trial name for batch/Job trials
+            job_name = trial_name
+
+            # Get logs from the job's pod
+            core_api = self.core_api 
+
+            # Find pods associated with the job
+            label_selector = f"katib.kubeflow.org/trial={trial_name}"
+            
+            # Check if trial has primaryPodLabels to identify the primary pod
+            primary_pod_labels = None
+            if (trial.get("status", {}).get("primaryPodLabels")):
+                primary_pod_labels = trial["status"]["primaryPodLabels"]
+                # Build label selector from primaryPodLabels
+                primary_selectors = [f"{k}={v}" for k, v in primary_pod_labels.items()]
+                label_selector = ",".join([label_selector] + primary_selectors)
+            
+            pods = core_api.list_namespaced_pod(
```

Comment:
you can directly do this
```suggestion
            pods = self.core_api.list_namespaced_pod(
```